test/perf/perf.psm1

```diff
@@ -32,19 +32,19 @@ function Start-Benchmarking
     .PARAMETER KeepFiles
     Indicates to keep all temporary files produced for running benchmarks.
     #>
-    [CmdletBinding(DefaultParameterSetName = 'Default')]
+    [CmdletBinding(DefaultParameterSetName = 'TargetFramework')]
     param(
-        [Parameter(ParameterSetName = 'Default')]
+        [Parameter(ParameterSetName = 'TargetPSVersion')]
         [ValidatePattern(
             '^7\.(0|1|2)\.\d+(-preview\.\d{1,2})?$',
```

Comment:
Defining the pattern in this way requires that it be updated for every new minor (and/or major) PSVersion. This seems like a not-particularly-great idea to me, is there not a way the function can check the supplied version against an automatically-determined list of possible versions?

Reply:
Yeah, the current validation is not optimal. In fact, not just here, the `powershell-perf.csproj` also requires manual updates when we move to a new PS version or new .NET runtime, because it hardcodes the supported .NET frameworks, as well as the latest PS version corresponding to a specific .NET framework. It would be nice if we can avoid the need of manual updates to both `perf.psm1` and `powershell-perf.csproj`.

Reply:
We could move this in a json config file.